### PR TITLE
Task vv/147: my-podcast-functionality

### PR DESCRIPTION
**Pull request by vitalik-prog**

### CODE REVIEW

**Review by yevhenprybytok — DISMISSED** [review]

File: backend/src/data/repositories/podcast/podcast.repository.ts [hunk 1]
```diff
@@ -33,8 +33,8 @@ class Podcast {
     return this.#PodcastModel.query().updateAndFetchById(id, payload);
   }
 
-  public getAllByUserId(userId: string): Promise<TPodcast[]> {
-    return this.#PodcastModel.query().where({ user_id: userId }).withGraphJoined('image');
+  public getAllByUserId(filterParams: { user_id: string }): Promise<TPodcast[]> {
```
Comment (artem-totality) [inline backend/src/data/repositories/podcast/podcast.repository.ts]: May be better don`t change getAllByUserId. This function has to return all users by id, and then to filter them  on next level in podcastService.

Comment (vitalik-prog) [reply backend/src/data/repositories/podcast/podcast.repository.ts]: if we have a large database, processing it with js will be very expensive. IMHO

File: backend/src/api/podcasts/podcasts.api.ts [hunk 2]
```diff
@@ -33,7 +33,7 @@ const initPodcastsApi = ({ apiRouter, podcastService }: Args): Router => {
     PodcastsApiPath.USERS_$ID,
     handleAsyncApi(async (req, res) => {
       return res
-        .send(await podcastService.getAllByUserId(req.params.id))
+        .send(await podcastService.getAllByUserId(req.params.id, String(req.user?.id)))
```
Comment (what1s1ove) [inline backend/src/api/podcasts/podcasts.api.ts]:
```suggestion
        .send(await podcastService.getAllByUserId(req.params.id, Number(req.user?.id)))
```

File: backend/src/middlewares/identify-user/identify-user.middleware.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { StrategyName } from '~/common/enums/enums';
+import { RequestHandler } from 'express';
+import passport from 'passport';
```
Comment (what1s1ove) [inline backend/src/middlewares/identify-user/identify-user.middleware.ts:2]: keep order

File: backend/src/data/repositories/podcast/podcast.repository.ts [hunk 4]
```diff
@@ -33,8 +33,8 @@ class Podcast {
     return this.#PodcastModel.query().updateAndFetchById(id, payload);
   }
 
-  public getAllByUserId(userId: string): Promise<TPodcast[]> {
-    return this.#PodcastModel.query().where({ user_id: userId }).withGraphJoined('[image, user]');
+  public getAllByUserId(filterParams: { user_id: string }): Promise<TPodcast[]> {
```
Comment (what1s1ove) [inline backend/src/data/repositories/podcast/podcast.repository.ts]: create a type for it

File: backend/src/middlewares/identify-user/identify-user.middleware.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+const identifyUser: RequestHandler = (req, res, next) => {
+  passport.authenticate(StrategyName.JWT, (err, user) => {
+    if (err) next(err);
```
Comment (what1s1ove) [inline backend/src/middlewares/identify-user/identify-user.middleware.ts]: check **B12** criteria

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/data/repositories/podcast/podcast.repository.ts [hunk 6]
```diff
@@ -6,6 +6,11 @@ import {
 import { PodcastType } from '~/common/enums/enums';
 import { PodcastModel as PodcastM } from '~/data/models/models';
 
+type filterParams = {
+  user_id: number;
+  type?: PodcastType.PUBLIC;
```
Comment (what1s1ove) [inline backend/src/data/repositories/podcast/podcast.repository.ts]:
```suggestion
  type?: PodcastType;
```

File: backend/src/data/repositories/podcast/podcast.repository.ts [hunk 7]
```diff
@@ -6,6 +6,11 @@ import {
 import { PodcastType } from '~/common/enums/enums';
 import { PodcastModel as PodcastM } from '~/data/models/models';
 
+type filterParams = {
```
Comment (what1s1ove) [inline backend/src/data/repositories/podcast/podcast.repository.ts]: add move readable name and move it to common

File: backend/src/services/podcast/podcast.service.ts [hunk 8]
```diff
@@ -173,8 +173,17 @@ class Podcast {
     return podcast;
   }
 
-  public getAllByUserId(userId: string): Promise<TPodcast[]> {
-    return this.#podcastRepository.getAllByUserId(userId);
+  public getAllByUserId(searchedUserId: number, requestUserId: number | undefined): Promise<TPodcast[]> {
+    const filterParams = {
```
Comment (what1s1ove) [inline backend/src/services/podcast/podcast.service.ts]: type it

File: backend/src/services/podcast/podcast.service.ts [hunk 9]
```diff
@@ -173,8 +173,17 @@ class Podcast {
...
+      user_id: searchedUserId,
+    };
+    const isRequestUserAuthorised = Boolean(requestUserId);
+
+    if (!isRequestUserAuthorised || searchedUserId !== requestUserId) {
+      Object.assign(filterParams, { type: PodcastType.PUBLIC });
```
Comment (what1s1ove) [inline backend/src/services/podcast/podcast.service.ts]:
```suggestion
      filterParams.type = PodcastType.PUBLIC;
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/common/types/podcast/podcasts-returned.type.ts [hunk 10]
```diff
@@ -0,0 +1,8 @@
+import { PodcastType } from '~/common/enums/enums';
+
+type PodcastsReturnedType = {
```
Comment (what1s1ove) [inline backend/src/common/types/podcast/podcasts-returned.type.ts]:
```suggestion
type UserPodcastQueryParams = {
```

Comment (what1s1ove) [reply backend/src/common/types/podcast/podcasts-returned.type.ts]: don't forget rename file

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**